nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSStateFlattener.java

```diff
@@ -37,17 +38,31 @@
 public class DBSStateFlattener {
 
     public static Map<String, Serializable> flatten(State state) {
+        return flatten(state, Collections.emptyMap());
+    }
+
+    /**
+     * Flatten with optional property mappings
+     * @param state State
+     * @param keyMappings A map of key/values to replace
```

Comment:
Javadoc style nitpicks: First sentence should end with a `.` and should be 3rd person declarative rather than 2nd person imperative (—> Flattens). `@param` and `@return` are not full sentences so no initial capital and no final `.`.